docs/extensions/readme.md

```diff
@@ -1173,6 +1174,112 @@ When set, specifies the header parameter to be used instead of `x-ms-client-requ
 
+## x-ms-allowed-arm-id-types
+
+Can only be set on `"type": "string"` fields with `"format": "arm-id"`.
+When set, specifies the set of resource types which can be referenced by this `arm-id`. If this extension isn't provided for a particular `arm-id`, the field can refer to any valid ARM ID.
+
+**Parent element**: [Parameter Object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#parameterObject), [Schema Object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#schemaObject), or [Items Object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#itemsObject)
+
+**Schema**:
+
+| Field Name    | Type       | Description                                      |
+| ------------- | ---------- | ------------------------------------------------ |
+| allowedTypes  | `[string]` | **Required** An array of allowed ARM resource ID kinds. Each element represents a particular type of ARM resource which can be referred to by this `arm-id`. See [Allowed formats](#allowed-arm-id-formats) for a detailed descrition of the allowed formats. |
+
+### Allowed ARM ID formats
+Note: When reading the formats below, `{}` is left as-is in the `allowedTypes` entry. Named fields such as `{provider}` must be given concrete values in the `allowedTypes` entry.
+
+| Resource kind                   | Format                                                                      | Example `allowedTypes` entry                                                       |
+| ------------------------------- | --------------------------------------------------------------------------- | ---------------------------------------------------------------------------------- |
+| Resource in a tenant            | `/providers/{provider}/{resourceType}`                                      | `/providers/Microsoft.Capacity/reservationOrders`                                  |
+| Resource in a subscription      | `/subscriptions/{}/providers/{provider}/{resourceType}`                     | `/subscriptions/{}/providers/Microsoft.Resources/resourceGroups`                   |
+| Resource in a resource group    | `/subscriptions/{}/resourceGroups/{}/providers/{provider}/{resourceType}`   | `/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/virtualNetworks`  |
+
+Sub-resources are specified in the same manner as their parent resource but with additional paths on the end. For example to refer to a subnet: `/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/virtualNetworks/{}/subnets`.
+
+**Example**: An `arm-id` field with no extension
+
+```json5
+"MyExampleType": {
+  "properties": {
+    "id": {
+      "type": "string",
+      "format": "arm-id"
+    }
+  }
+}
+```
+
+**Example**: An `arm-id` field that must refer to a virtual network
+```json5
+"MyExampleType": {
+  "properties": {
+    "vnetId": {
+      "type": "string",
+      "format": "arm-id",
+      "x-ms-allowed-arm-id-types": {
+        "allowedTypes": [
```

Comment:
So we need this nesting for this new field?

As `x-ms-allowed-arm-id-types` and `allowedTypes` seems a bit of a duplication of names.

Reply:
Yeah agree it's a bit duplicate.

I made it a complex type rather than a raw array just because it felt more extensible if in the future we wanted like a `deniedTypes` or something too (i.e. all but this?).

If folks are confident on the shape I've proposed here I'm happy to collapse this to just a simple array

Reply:
Maybe we could change it to `x-ms-arm-id-types` with an `allowed` field?

Reply:
Changed it to `x-ms-arm-id-details`